src/atomicdex/models/qt.addressbook.contact.addresses.model.cpp

```diff
@@ -0,0 +1,186 @@
+    }
+    
+    bool addressbook_contact_addresses_model::insertRows(int position, int rows, const QModelIndex& parent)
+    {
+        beginInsertRows(parent, position, position + rows);
+        for (int row = 0; row < rows; row++)
+        {
+            m_model_data.push_back({"", ""});
+        }
+        endInsertRows();
+        return true;
+    }
+
+    bool addressbook_contact_addresses_model::removeRows(int position, int rows, const QModelIndex& parent)
+    {
+        auto& addrbook_manager = m_system_manager.get_system<addressbook_manager>();
+        
+        beginRemoveRows(parent, position, position + rows);
+        for (int row = 0; row < rows; row++)
+        {
+            const auto& data = m_model_data.at(position);
+    
+            addrbook_manager.remove_contact_wallet_info(m_name.toStdString(), m_type.toStdString(), data.key.toStdString());
+            m_model_data.removeAt(position);
+        }
+        endRemoveRows();
+        return true;
+    }
+    
+    QHash<int, QByteArray> addressbook_contact_addresses_model::roleNames() const
+    {
+        return {{TypeRole, "type"}, {KeyRole, "key"}, {AddressRole, "value"}};
+    }
+}
+
+//! QML API implementation.
+namespace atomic_dex
+{
+    void addressbook_contact_addresses_model::add_address_entry()
+    {
+        insertRow(rowCount());
+    }
+    
+    void addressbook_contact_addresses_model::remove_address_entry(int row)
+    {
+        removeRow(row);
+    }
+    
+    void addressbook_contact_addresses_model::remove_address_entries()
+    {
+        removeRows(0, rowCount(), QModelIndex());
+    }
+    
+    const QString& addressbook_contact_addresses_model::get_type() const noexcept
+    {
+        return m_type;
+    }
+}
+
+namespace atomic_dex
+{
+    void addressbook_contact_addresses_model::populate()
+    {
+        const auto& addrbook_manager = m_system_manager.get_system<addressbook_manager>();
+        
+        if (!addrbook_manager.has_wallet_info(m_name.toStdString(), m_type.toStdString()))
+        {
+            return;
+        }
+        
+        const auto& addresses = addrbook_manager.get_wallet_info(m_name.toStdString(), m_type.toStdString()).at("addresses");
+        
+        beginInsertRows(QModelIndex(), 0, addresses.size());
+        for (auto it = addresses.begin(); it != addresses.end(); ++it)
+        {
+            m_model_data.push_back(address{.key = QString::fromStdString(it.key()), .value = QString::fromStdString(it.value())});
+        }
+        endInsertRows();
+    }
+    
+    void addressbook_contact_addresses_model::save()
+    {
+        auto& addrbook_manager = m_system_manager.get_system<addressbook_manager>();
+        
+        // Cleans existing wallet info persistent data before.
+        if (addrbook_manager.has_wallet_info(m_name.toStdString(), m_type.toStdString()))
+        {
+            addrbook_manager.remove_contact_wallet_info(m_name.toStdString(), m_type.toStdString());
+        }
```

Comment:
Is not better to have a `remove_if` instead of `O(n)` algorithm to check and `O(n)` algorithm to remove ?

you probably want a `std::remove_if` equivalent check if `nlohmann::json` already support it